packages/core/src/StateMachine.ts

```diff
@@ -338,15 +351,27 @@ export class StateMachine<
   /**
    * Returns the initial `State` instance, with reference to `self` as an `ActorRef`.
    */
-  public getInitialState(): State<TContext, TEvent, TResolvedTypesMeta> {
-    const { preInitialState } = this;
-    const nextState = resolveMicroTransition(this, [], preInitialState);
+  public getInitialState(
+    actorCtx?: ActorContext<TEvent, State<TContext, TEvent>>
+  ): State<TContext, TEvent, TResolvedTypesMeta> {
+    const preInitialState = this.getPreInitialState(actorCtx);
+    const nextState = resolveMicroTransition(
+      this,
+      [],
+      preInitialState,
+      actorCtx
+    );
     nextState.actions.unshift(...preInitialState.actions);
 
-    const macroState = macrostep(nextState, null as any, this) as StateFrom<
-      typeof this
-    >;
+    // TODO: remove use of null; use xstate.init instead
```

Comment:
This feels like a very important thing that should be handled before merging this PR.